rootfs/etc/nginx/lua/sticky.lua

```diff
@@ -0,0 +1,118 @@
+local json = require('cjson')
+local str = require("resty.string")
+local sha1_crypto = require("resty.sha1")
+local md5_crypto = require("resty.md5")
+
+local sticky_sessions = ngx.shared.sticky_sessions
+
+local DEFAULT_SESSION_COOKIE_NAME = "route"
+local DEFAULT_SESSION_COOKIE_HASH = "md5"
+-- Currently STICKY_TIMEOUT never expires
+local STICKY_TIMEOUT = 0
+
+local _M = {}
+
+local function md5_digest(raw)
+  local md5 = md5_crypto:new()
+  if not md5 then
+    ngx.say("failed to create the md5 object")
+    return
+  end
+  md5:update(raw)
+  return str.to_hex(md5:final())
```

Comment:
same as above, what if `:final` fails

Reply:
In all of the examples there is no error handling for `*:final()`

https://github.com/openresty/lua-resty-string